streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1084,12 +1088,15 @@ private boolean populateActiveTaskAndPartitionsLists(final List<TopicPartition>
                 // If the partition is new to this consumer but is still owned by another, remove from the assignment
                 // until it has been revoked and can safely be reassigned according to the COOPERATIVE protocol
                 if (newPartitionForConsumer && allOwnedPartitions.contains(partition)) {
-                    log.info("Removing task {} from assignment until it is safely revoked in followup rebalance", taskId);
-                    clientState.unassignActive(taskId);
```

Comment:
By the way, this tripped me up quite a bit. Changing the clientState has absolutely no effect at this point. Might be nice to enforce that in code, but I'd like to defer it (I took a stab, but it's a lot of changes).

Reply:
I guess it was nice to have as a sanity check that the task really was assigned to this client, and that we haven't already tried to remove it or something. But maybe we can take that for granted...? (I only mention it because it helped me find a bug in the thread-level stickiness PR)

Reply:
Ah, I see. I'll put it back with a comment that we're just using it for the internal assertions.